File: compiled/construct/integers.py
from construct import *
from construct.lib import *

integers = Struct(
	'magic1' / Const(b"\x50\x41\x43\x4B\x2D\x31"),
	'uint8' / Int8ub,
	'sint8' / Int8sb,
	'magic_uint' / Const(b"\x50\x41\x43\x4B\x2D\x55\x2D\x44\x45\x46"),
	'uint16' / Int16ul,
	'uint32' / Int32ul,
	'uint64' / Int64ul,
	'magic_sint' / Const(b"\x50\x41\x43\x4B\x2D\x53\x2D\x44\x45\x46"),
	'sint16' / Int16sl,
	'sint32' / Int32sl,
	'sint64' / Int64sl,
	'magic_uint_le' / Const(b"\x50\x41\x43\x4B\x2D\x55\x2D\x4C\x45"),
	'uint16le' / Int16ul,
	'uint32le' / Int32ul,
	'uint64le' / Int64ul,
	'magic_sint_le' / Const(b"\x50\x41\x43\x4B\x2D\x53\x2D\x4C\x45"),
	'sint16le' / Int16sl,
	'sint32le' / Int32sl,
	'sint64le' / Int64sl,
	'magic_uint_be' / Const(b"\x50\x41\x43\x4B\x2D\x55\x2D\x42\x45"),
	'uint16be' / Int16ub,
	'uint32be' / Int32ub,
	'uint64be' / Int64ub,
	'magic_sint_be' / Const(b"\x50\x41\x43\x4B\x2D\x53\x2D\x42\x45"),
	'sint16be' / Int16sb,
	'sint32be' / Int32sb,
	'sint64be' / Int64sb,
)

_schema = integers

File: compiled/nim/integers.nim
import kaitai_struct_nim_runtime
import options

template defineEnum(typ) =
  type typ* = distinct int64
  proc `==`*(x, y: typ): bool {.borrow.}

type
  Integers* = ref object of KaitaiStruct
    magic1*: seq[byte]
    uint8*: uint8
    sint8*: int8
    magicUint*: seq[byte]
    uint16*: uint16
    uint32*: uint32
    uint64*: uint64
    magicSint*: seq[byte]
    sint16*: int16
    sint32*: int32
    sint64*: int64
    magicUintLe*: seq[byte]
    uint16le*: uint16
    uint32le*: uint32
    uint64le*: uint64
    magicSintLe*: seq[byte]
    sint16le*: int16
    sint32le*: int32
    sint64le*: int64
    magicUintBe*: seq[byte]
    uint16be*: uint16
    uint32be*: uint32
    uint64be*: uint64
    magicSintBe*: seq[byte]
    sint16be*: int16
    sint32be*: int32
    sint64be*: int64
    parent*: KaitaiStruct

proc read*(_: typedesc[Integers], io: KaitaiStream, root: KaitaiStruct, parent: KaitaiStruct): Integers


proc read*(_: typedesc[Integers], io: KaitaiStream, root: KaitaiStruct, parent: KaitaiStruct): Integers =
  template this: untyped = result
  this = new(Integers)
  let root = if root == nil: cast[Integers](this) else: cast[Integers](root)
  this.io = io
  this.root = root
  this.parent = parent

  this.magic1 = this.io.readBytes(int(6))
  this.uint8 = this.io.readU1()
  this.sint8 = this.io.readS1()
  this.magicUint = this.io.readBytes(int(10))
  this.uint16 = this.io.readU2le()
  this.uint32 = this.io.readU4le()
  this.uint64 = this.io.readU8le()
  this.magicSint = this.io.readBytes(int(10))
  this.sint16 = this.io.readS2le()
  this.sint32 = this.io.readS4le()
  this.sint64 = this.io.readS8le()
  this.magicUintLe = this.io.readBytes(int(9))
  this.uint16le = this.io.readU2le()
  this.uint32le = this.io.readU4le()
  this.uint64le = this.io.readU8le()
  this.magicSintLe = this.io.readBytes(int(9))
  this.sint16le = this.io.readS2le()
  this.sint32le = this.io.readS4le()
  this.sint64le = this.io.readS8le()
  this.magicUintBe = this.io.readBytes(int(9))
  this.uint16be = this.io.readU2be()
  this.uint32be = this.io.readU4be()
  this.uint64be = this.io.readU8be()
  this.magicSintBe = this.io.readBytes(int(9))
  this.sint16be = this.io.readS2be()
  this.sint32be = this.io.readS4be()
  this.sint64be = this.io.readS8be()

proc fromFile*(_: typedesc[Integers], filename: string): Integers =
  Integers.read(newKaitaiFileStream(filename), nil, nil)

